### PR TITLE
RGB_Printer_Events---Updated

### PR DESCRIPTION
**Pull request by Tannoo**
Corrections and updates from RCBugFix.

### CODE REVIEW

File: Marlin/Marlin_main.cpp [hunk 1]
```diff
@@ -6616,9 +6788,11 @@ inline void gcode_M109() {
 
     #if ENABLED(PRINTER_EVENT_LEDS)
       // Gradually change LED strip from violet to red as nozzle heats up
-      if (!wants_to_cool) {
-        const uint8_t blue = map(constrain(temp, start_temp, target_temp), start_temp, target_temp, 255, 0);
-        if (blue != old_blue) set_led_color(255, 0, (old_blue = blue));
+      if (wait_for_heatup && !wants_to_cool) {
```
Comment (thinkyhead) [inline Marlin/Marlin_main.cpp]: `wait_for_heatup` is always set during the loop, so can be omitted here.

File: Marlin/Marlin_main.cpp [hunk 2]
```diff
@@ -6616,9 +6788,11 @@ inline void gcode_M109() {
...
+        uint8_t blue = map(constrain(temp, start_temp, target_temp), start_temp, target_temp, 255, 0);
+        if (blue == old_blue) set_rgb_color(255, 0, 255, 0, 0, 1);   //Purple to start
+        if (blue != old_blue) set_rgb_color(255, 0, blue, 0, 0, 1);  //Start transitioning to Red
```
Comment (thinkyhead) [inline Marlin/Marlin_main.cpp]: These lines resurrect the bad logic I pointed out earlier. Please just extend add the new parameters to the existing `set_led_color` line. Let's preserve the `set_led_color` function name and keep it as the main setter.

**Review by thinkyhead — CHANGES_REQUESTED** [review]
Various formatting and logical changes are needed. I will see about patching this up in a day or two.

File: Marlin/Marlin_main.cpp (hunk 1 above)
Comment (Tannoo) [reply Marlin/Marlin_main.cpp]: Done

File: Marlin/Marlin_main.cpp (hunk 2 above)
Comment (Tannoo) [reply Marlin/Marlin_main.cpp]: Changed it to set_led_color.

File: Marlin/Configuration.h [hunk 3]
```diff
@@ -1535,13 +1535,51 @@
...
+ *           *****************************************************************
+ *           **NOTE: This feature requires FastLED library to be installed. **
+ *           **         https://github.com/FastLED/FastLED.git              **
+ *           *****************************************************************
+ */
+//#define LEDSTRIP
+
+#if ENABLED(LEDSTRIP)
+
+      /* Strip specific items */
```
Comment (thinkyhead) [inline Marlin/Configuration.h]: delete this line and the blank line above it

**Review by thinkyhead — CHANGES_REQUESTED** [review]
minor tweak to the config additions

**Comment by thinkyhead** [comment]
This is on hold until after release. We should revisit this and see about making it a sub-feature of the existing RGB LED stuff instead of making it a whole separate feature.

**Comment by Tannoo** [comment]
Well, my end doesn't show any conflicts. Online OR local.
I click on the `Resolve conflicts` button and I can see the conflicts. That is the only place I can see them and the online editor doesn't work for me.

Let me get this caught up with all the recent changes.

**Comment by Tannoo** [comment]
There we go.

File: Marlin/Configuration.h (hunk 3 above)
Comment (Tannoo) [reply Marlin/Configuration.h]: Done.

**Comment by Tannoo** [comment]
Yeah, I found this issue also and is fixed in this PR.

http://forums.reprap.org/read.php?415,766651

**Pull request closed without merging**